airbyte-integrations/connectors/source-bigquery/source_bigquery/streams.py

```diff
@@ -0,0 +1,187 @@
+#
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+#
+
+
+import logging
+from abc import ABC
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Tuple
+
+import requests
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http import HttpStream, HttpSubStream
+from airbyte_cdk.sources.streams.http.auth import TokenAuthenticator
+from airbyte_protocol.models import SyncMode
+from airbyte_cdk.models import AirbyteCatalog, AirbyteMessage, AirbyteStateMessage, ConfiguredAirbyteCatalog
+from airbyte_cdk.logger import AirbyteLogger
+
+"""
+This file provides a stubbed example of how to use the Airbyte CDK to develop both a source connector which supports full refresh or and an
+incremental syncs from an HTTP API.
+
+The various TODOs are both implementation hints and steps - fulfilling all the TODOs should be sufficient to implement one basic and one incremental
+stream from a source. This pattern is the same one used by Airbyte internally to implement connectors.
+
+The approach here is not authoritative, and devs are free to use their own judgement.
+
+There are additional required TODOs in the files within the integration_tests folder and the spec.yaml file.
+"""
+URL_BASE: str = "https://bigquery.googleapis.com"
+    
+
+class BigqueryStream(HttpStream, ABC):
+    """
+    """ 
+    url_base = URL_BASE
+    primary_key = "id"
+    raise_on_http_errors = True
+
+    def __init__(self, stream_path: str, stream_name: str, stream_schema, **kwargs):
+        super().__init__(**kwargs)
+        self.stream_path = stream_path
+        self.stream_name = stream_name
+        self.stream_schema = stream_schema
+
+    @property
+    def name(self):
+        return self.stream_name
```

Comment:
Given the current setup, if the user has two BigqueryTables to sync, both will have the same stream name which is not valid